### PR TITLE
New readback

### PR DESCRIPTION
**Pull request by racs4**

### CODE REVIEW

**Comment by VictorTaelin** [comment]
If this is the proper readback function (that resolves `dups`), then it should be called just `readback_term`, not `readback_linear_term`. I love how you manage to convert these recursive functions to stack-based procedures without losing the structure, very cool.

**Comment by racs4** [comment]
[Here](https://gist.github.com/racs4/6ec62ee61dad751a675901bf6dfd02c8) are some terms before and after readback (some of them also has a comparison with the old one)

**Comment by VictorTaelin** [comment]
Just to be sure: this is the same algorithm as the `as_term` function on `hvm/readback.rs`? If so, eventually we would like to update that one too, to avoid using stacks there.

**Comment by racs4** [comment]
Yes, it is the same as `as_term` function.

File: src/hvm.rs [hunk 1]
```diff
@@ -3790,198 +3790,221 @@ fn show_runtime_error(err: RuntimeError) -> String {
...
+
     let mut output = Vec::new();
+    let mut stack = vec![StackItem::Term(term)];
+
     while !stack.is_empty() {
       let item = stack.pop().unwrap();
       match item {
-        StackItem::Resolver(term) => {
+        StackItem::Term(term) => {
+          debug_assert!(term != 0);
```
Comment (steinerkelvin) [inline src/hvm.rs:3879]: It would be nice to add this check to some other functions while we don't refactor to a newtype.

**Pull request merged**